### PR TITLE
Add bash completion to bindist

### PR DESCRIPTION
**Pull request by hseg**
From a cursory check, this appears to be what's needed, but I may be mistaken.
Would appreciate a review

Closes: #986

### CODE REVIEW

**Comment by hseg** [comment]
Oops, misread the workflow -- one still needs to copy completion to `tmp` first.
Let me quickly fix that...

File: .github/workflows/binaries-linux-arm32v7.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
           docker cp $container_id:/root/hledger tmp
           docker rm -v $container_id
           cd tmp
-          tar cvf hledger-linux-arm32v7.tar hledger hledger-ui hledger-web
+          tar cvf hledger-linux-arm32v7.tar hledger hledger-ui hledger-web hledger/shell-completion/hledger-completion.bash
```
Comment (simonmichael) [inline .github/workflows/binaries-linux-arm32v7.yml:37]: This looks wrong - is `hledger` the binary file or a directory here ?

Comment (hseg) [reply .github/workflows/binaries-linux-arm32v7.yml:37]: Was meant to be the directory

**Comment by simonmichael** [comment]
And thanks for making a start on this. CI scripts are hard to test and debug. I guess the alternative is to embed shell completions in the hledger binaries and add a flag that writes them to stdout ? Is that the standard/preferred idiom, or just unnecessary complication ?

**Comment by hseg** [comment]
I know projects using `optparse-applicative` do this, cf eg https://gitlab.archlinux.org/archlinux/packaging/packages/stack/-/blame/main/PKGBUILD#L84
But that appeared imo to be more work, especially after I'd tracked down your releng.

**Comment by hseg** [comment]
Hrm. Perhaps by adding the completion to `extra-source-files`, it'll be copied somewhere useful by `stack`?

**Comment by hseg** [comment]
further oops -- just noticed I've opened this PR from my master branch, without bothering to create a new branch. merging this might end up being messy, sorry for that.

**Comment by hseg** [comment]
Yeah, I'm not seeing anything there.
However... before the move to `tmp` to construct the tarball, aren't we at the repository root? Couldn't we just copy the completion in there?

**Comment by hseg** [comment]
I think this might do the trick -- I note that `binaries-linux-x64.yml` does something similar.

**Comment by hseg** [comment]
I've got to leave in a minute, I'll pick this up again tomorrow.

**Comment by hseg** [comment]
I tried testing locally using https://github.com/nektos/act, but I don't have the github tokens. Would you mind testing this?

**Comment by simonmichael** [comment]
I've broadened the scope of #986 a little and posted some problem reports there. I'll get back to this issue once I can see completions working on my machine.

File: .github/workflows/binaries-linux-arm32v7.yml (hunk 1 above)
Comment (simonmichael) [reply .github/workflows/binaries-linux-arm32v7.yml:37]: @hseg, all the other changes look right, but am I right in thinking this one is still wrong ? We rarely use this workflow, but as written it seems to treat `hledger` as both the binary and the directory at the same time.

**Comment by hseg** [comment]
Yeah, it doesn't follow the same flow as the others, which is a smell. You want to add a `docker cp` line to copy out the shell completions into `tmp` as well.
In theory, this means you could write the `tar` line as just
```bash
tar cvf hledger-linux-arm32v7.tar *
```
Don't know how I missed it.

BTW, after my prodding, the AUR maintainer also added the manpages to the package on their own, but the same logic goes for them as well - they're user-facing content, they should be in the bindist.

**Comment by simonmichael** [comment]
Thanks for helping with packaging and distribution @hseg. I wasn't really aware of packagers using the release binaries from github; now I am. I have merged this and we'll keep improving things. Perhaps the "release bindists" should have all the same extra source files that the hackage tarballs have.

**Comment by hseg** [comment]
No problem. Though I notice you didn't change the linux-arm32v7 workflow you were asking about - won't this create a different directory structure in the tarball from the others? I'm afk, so I can't test this right now.

**Comment by simonmichael** [comment]
Yup, I’ll follow up on that.

**Pull request merged**